modules/stochastic_tools/include/state_sim/common/StateSimIdAndDesc.h

```diff
@@ -0,0 +1,30 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMIDANDDESC_H
+#define STATESIMIDANDDESC_H
+
+#include <string>
+
+/**
+ * Simple class to store the path of state movement and the reason for that movement.
+ */
+class StateSimIdAndDesc
+{
+public:
+  /**
+   * This is the main construter for the user.
+   * @param in_id - item id
+   * @param in_desc - description.
+   */
+  StateSimIdAndDesc(int in_id, std::string in_desc) : id(in_id), desc(in_desc) {};
```

Comment:
I would like it to be a struct, but I also want something like a constructor.  Is there a better way to get this effect?